newton/_src/usd/utils.py

```
@@ -2449,6 +2564,69 @@ def _extract_preview_surface_properties(shader: UsdShade.Shader | None, prim: Us
+    - A connected ``UsdUVTexture``: accepted only when the connected output is a
+      multi-channel color output (``rgb`` / ``rgba``). Single-channel outputs
+      (``r``/``g``/``b``/``a``) are scalar data maps (roughness, metallic, ...)
+      and are ignored, so a data map is never mistaken for the diffuse texture —
+      independent of input naming.
+    - A direct asset value (e.g. an MDL ``diffuse_texture`` parameter): there is
+      no texture node to inspect, so the base-color input is identified by its
+      (well-known) name.
+    """
+    try:
+        producing = UsdShade.Utils.GetValueProducingAttributes(surface_input)
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)